build/dsc/templates/dsc_type.rb.erb

```diff
@@ -103,6 +103,8 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
     defaultto { :<%= begin resource.ensure_value rescue 'present' end %> }
   end
 
+  autonotify(:reboot) { ['dsc_reboot'] }
```

Comment:
We need to do the Puppet version check here before using the autonotify.

We do a similar check in:
https://github.com/puppetlabs/puppetlabs-acl/blob/b1bd084fe7a43c99dbe44067a4898171b9fe6cf1/lib/puppet/util/monkey_patches.rb#L134